src/Tasks/FileIO/GetFileHash.cs

```diff
@@ -67,26 +75,33 @@ public override bool Execute()
                 return false;
             }
 
-            foreach (var file in Files)
+            Parallel.For(0, Files.Length, index =>
             {
+                var file = Files[index];
+
                 if (!FileSystems.Default.FileExists(file.ItemSpec))
                 {
                     Log.LogErrorWithCodeFromResources("FileHash.FileNotFound", file.ItemSpec);
+                    return;
                 }
-            }
+
+                var hash = ComputeHash(algorithmFactory, file.ItemSpec);
+                var encodedHash = EncodeHash(encoding, hash);
+
+                lock (file)
```

Comment:
nit: Locking on public objects should generally be avoided.

Reply:
Fixed